hnn_core/network_builder.py

```diff
@@ -644,19 +648,22 @@ def _record_spikes(self):
             if _PC.gid_exists(gid):
                 _PC.spike_record(gid, self._spiketimes, self._spikegids)
 
-    # aggregate recording all the somatic voltages for pyr
     def aggregate_currents(self):
-        """This method must be run post-integration."""
-        # this is quite ugly
+        """Aggregate somatic currents for Pyramidal cells."""
+        for cell in self.cells:
+            if cell.celltype in ('L5_pyramidal', 'L2_pyramidal'):
+                # iterate over dict_currents created in
+                # cell.record_current_soma()
+                for _, I_soma in cell.dict_currents.items():
+                    # self.current['L5_pyramidal_soma'] was created upon
+                    # in parallel, each node has its own NetworkBuilder()
+                    self.current['%s_soma' % cell.celltype].add(I_soma)
```

Comment:
Do you know where `self.current` gets used outside of `network_builder.py`? Honestly it seems like the data flow reaches a dead end here which is the same situation as not being able to disable `vsoma` recording in #190.